### PR TITLE
tests implemented

### PR DESCRIPTION
**Pull request by katy-mozharovska**

### CODE REVIEW

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 1]
```diff
@@ -1,14 +1,55 @@
...
-        return null;
+        validateUser(user);
+        return storageDao.add(user);
+    }
+
+    private void validateUser(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserException("User with login '"
+                    + user.getLogin() + "' already exists.");
+        }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:24]: put this after two next checks

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 2]
```diff
@@ -1,14 +1,55 @@
...
+
+        if (user.getLogin() == null) {
+            throw new InvalidUserException("Login is required");
+        }
+
+        if (user.getLogin().length() < 6) {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:30]: use your constant

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 3]
```diff
@@ -1,14 +1,55 @@
...
+            throw new InvalidUserException("Invalid password '" + user.getPassword()
+                    + "'. Password must be at least '"
+                    + MIN_PASSWORD_LENGTH + "' characters long.");
+        }
+
+        if (user.getAge() == null) {
+            throw new InvalidUserException("Age is required");
+        }
+
+        if (user.getAge() < 18 || user.getAge() > 2147483647) {
```
Comment (okuzan) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:50]: use your constant, and better omit max limit check
image

```suggestion
        if (user.getAge() < MIN_AGE) {
```

**Review by okuzan — CHANGES_REQUESTED** [review]
You need only one positive test, where every value is valid and everything passes ok, you can keep one with multiple registrations if you wish. Short login, password - make sure to test precisely the edge case (6), and the value right below it (5), for the age - 18 and 17 (you have these). Also, it's more common to use lowercase after _underscore. Your private utility method is a constructor, so use a constructor instead. You can have a default valid user as a field, set it up in before each, and amend necessary fields for each test (set to short password for short password test. other values remain valid)

**Review by okuzan — APPROVED** [review]
